cmd/lakectl/cmd/doctor.go

```diff
@@ -0,0 +1,99 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/treeverse/lakefs/pkg/api"
+)
+
+var (
+	ErrCredential       = errors.New("credential error")
+	ErrWrongEndpointURI = errors.New("wrong endpoint_uri error")
+)
+
+// doctorCmd represents the doctor command
+var doctorCmd = &cobra.Command{
+	Use:   "doctor",
+	Short: "Run a basic diagnosis of the LakeFS configuration",
+	Run: func(cmd *cobra.Command, args []string) {
+		err := ListRepositoriesAndAnalyze(cmd.Context())
+		if err == nil {
+			Fmt("LakeFS doctor could not find any configuration issues\n")
+			return
+		}
+
+		configFileName := viper.GetViper().ConfigFileUsed()
+		Fmt("It looks like you have a problem with your `%v` file.\n", configFileName)
+
+		switch {
+		case errors.Is(err, ErrCredential):
+			Fmt("It is possible that the `access_key_id` or `secret_access_key` you supplied are wrong.\n")
+		case errors.Is(err, ErrWrongEndpointURI):
+			Fmt("Probably your endpoint url is wrong.\n")
+		}
+
+		accessKeyID := cfg.Values.Credentials.AccessKeyID
+		if !IsValidAccessKeyID(accessKeyID) {
+			Fmt("access_key_id value looks suspicious: %v\n", accessKeyID)
+		}
+
+		secretAccessKey := cfg.Values.Credentials.SecretAccessKey
+		if !IsValidSecretAccessKey(secretAccessKey) {
+			Fmt("secret_access_key value looks suspicious...\n")
+		}
+
+		serverEndpoint := cfg.Values.Server.EndpointURL
+		if !strings.HasSuffix(serverEndpoint, api.BaseURL) {
+			Fmt("Suspicious URI format for server.endpoint_url: %v, doesn't have `%v` suffix.\n", serverEndpoint, api.BaseURL)
+		}
+	},
+}
+
+func ListRepositoriesAndAnalyze(ctx context.Context) error {
+	serverEndpoint := cfg.Values.Server.EndpointURL
+	_, err := url.Parse(serverEndpoint)
```

Comment:
I moved this validation here since `getClient` method dies on `url.Parse` error

Reply:
This might be better phrased as a comment in the code -- so I don't "clean it up" in 3 months:
```suggestion
	# getClient dies on url.Parse error, so check it first.
	_, err := url.Parse(serverEndpoint)
```